app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -488,8 +497,22 @@ def parse_args(args)
 
       args.except!(:hardware_version) if args[:hardware_version] == 'Default'
 
-      firmware_type = args.delete(:firmware_type)
-      args[:firmware] = firmware_mapping(firmware_type) if args[:firmware] == 'automatic'
+      firmware_type = args.delete(:firmware_type).to_s
+
+      if args[:firmware] == 'automatic'
+        if firmware_type == 'none' || firmware_type.empty?
+          args[:firmware] = 'bios'
+        else
+          args[:firmware] = firmware_type
+        end
+      end
+
+      if args[:firmware]&.start_with?('uefi')
+        args[:secure_boot] = true if args[:firmware] == 'uefi_secure_boot'
+        args[:firmware] = 'efi'
+      end
+
+      args[:virtual_tpm] = ActiveRecord::Type::Boolean.new.cast(args[:virtual_tpm])
```

Comment:
I still think we can raise an exception here:
```suggestion
      args[:virtual_tpm] = ActiveRecord::Type::Boolean.new.cast(args[:virtual_tpm])
      if args[:virtual_tpm] && args[:firmware] != 'efi'
        raise ArgumentError, _('TPM is incompatible with BIOS firmware. Change firmware or disable TPM.')
      end
```